tests/unit/datastream_tests.cpp

```diff
@@ -385,6 +385,27 @@ EOSIO_TEST_BEGIN(datastream_stream_test)
    ds >> str;
    CHECK_EQUAL( cstr, str )
 
+   // -----------
+   // std::basic_string
+   ds.seekp(0);
+   fill(begin(datastream_buffer), end(datastream_buffer), 0);
+   static const std::basic_string<uint8_t> inputBasicString {0, 1, 2, 3, 4, 5};
```

Comment:
please add test with 2+ bytes type. something like `wchar_t`

Reply:
That was really, really good idea.
It occurred that the second solution with datastream.read/write was wrong. Probably because casting to `char*` was narrowing 2 bytes to 1 byte?
I reverted solution to the original one and it looks good.
Thanks!

Reply:
I think your issue was that you had `* sizeof(T)` in `read` but not in `write`. I can't think of other reason but if wouldn't work, try to use `static_cast` instead of `c-cast`. We have `std::string` that is using `read/write`. I propose to have consistent code. I see that -o2 supposed to make no difference here but anyway that is performance critical code so I think we should use `read/write` to make even non-optimized code faster.

Reply:
My code is copied from vector serializing so I believe it is consistent (because basically basic_string is a container).
What is the purpose of making non-optimized code faster? The difference would be only if someone deliberately builds code with debug flags. If it really is important then we need to create issues for improving both vectors and arrays serialization.

Reply:
> I think your issue was that you had `* sizeof(T)` in `read` but not in `write`.

Thanks, that was the issue.
I additionally changed `char*` to `void*` in read method, so I could avoid unnecessary `reinterpret_cast<char*>` from `wchar_t*` in my code (which shouldn't be needed anyway, because `memcpy` needs `void*` type).

Reply:
Short answer - I don't know for sure. My comment was based on the fact that cdt is not most recently installed compiler. this is our own patched compiler and its main purpose is to compile to web assembly. So I think it is easier to make it just optimal than check this behavior in wasm.
Regarding having optimal debug build - that may be helpful to avoid timeouts while we debug. You may recall errors in DUNE due to performance. Imagine you have slow machine and debug builds, it will be even slower so you need additional configuration to allow big timeouts